### PR TITLE
Add a param to getEventReadUpTo to have it ignore implicit read receipts

### PR DESCRIPTION
**Pull request by dbkr**
Store real receipts separately to make this work.

### CODE REVIEW

File: lib/models/room.js [hunk 1]
```diff
@@ -904,7 +914,7 @@ Room.prototype.getReceiptsForEvent = function(event) {
  * Add a receipt event to the room.
  * @param {MatrixEvent} event The m.receipt event.
  */
-Room.prototype.addReceipt = function(event) {
+Room.prototype.addReceipt = function(event, fake) {
```
Comment (richvdh) [inline lib/models/room.js]: could do with doc for `fake` param

File: lib/models/room.js [hunk 2]
```diff
@@ -877,17 +879,25 @@ Room.prototype.getUsersReadUpTo = function(event) {
  * Get the ID of the event that a given user has read up to, or null if we
  * have received no read receipts from them.
  * @param {String} userId The user ID to get read receipt event ID for
+ * @param {Boolean} ignoreSynthesized If true, do return only receipts that have been
```
Comment (richvdh) [inline lib/models/room.js]: s/do //

**Comment by richvdh** [comment]
What about the receipt synthesized in `_addLocalEchoReceipt`?

This feels slightly overspecced given we only care about the distinction for our own read receipts, but I don't really have any concrete suggestions for doing it better.

**Comment by dbkr** [comment]
I think I'm going with local echo receipts should not be returned if you ask for only real ones, although I'm not totally certain. I think we do want this mode to just give the server's version of the truth without any made-up stuff by the js sdk, since the receipt we sent might never actually get through to the server.

ptal.

**Comment by richvdh** [comment]
lgtm

**Pull request merged**